Library/ViewModels/PledgePaymentMethodsViewModelTests.swift

```diff
@@ -7,48 +7,72 @@ import ReactiveExtensions_TestHelpers
 final class PledgePaymentMethodsViewModelTests: TestCase {
   private let vm: PledgePaymentMethodsViewModelType = PledgePaymentMethodsViewModel()
 
+  private let applePayButtonHidden = TestObserver<Bool, Never>()
   private let notifyDelegateLoadPaymentMethodsError = TestObserver<String, Never>()
   private let reloadPaymentMethods = TestObserver<[GraphUserCreditCard.CreditCard], Never>()
 
   override func setUp() {
     super.setUp()
 
+    self.vm.outputs.applePayButtonHidden.observe(self.applePayButtonHidden.observer)
     self.vm.outputs.notifyDelegateLoadPaymentMethodsError
       .observe(self.notifyDelegateLoadPaymentMethodsError.observer)
     self.vm.outputs.reloadPaymentMethods.observe(self.reloadPaymentMethods.observer)
   }
 
-  func testReloadPaymentMethods_LoggedIn() {
+  func testReloadPaymentMethods_LoggedIn_ApplePayCapable_isFalse() {
     let response = UserEnvelope<GraphUserCreditCard>(me: GraphUserCreditCard.template)
     let mockService = MockService(fetchGraphCreditCardsResponse: response)
 
     withEnvironment(apiService: mockService, currentUser: User.template) {
       self.reloadPaymentMethods.assertDidNotEmitValue()
+      self.applePayButtonHidden.assertDidNotEmitValue()
 
-      self.vm.inputs.configureWith(User.template)
+      self.vm.inputs.configureWith((User.template, Project.template, false))
       self.vm.inputs.viewDidLoad()
 
       self.scheduler.run()
 
+      self.applePayButtonHidden.assertValues([true])
       self.reloadPaymentMethods.assertValue(response.me.storedCards.nodes)
     }
   }
 
-  func testReloadPaymentMethods_Error_LoggedIn() {
+  func testReloadPaymentMethods_LoggedIn_ApplePayCapable_isTrue() {
+    let response = UserEnvelope<GraphUserCreditCard>(me: GraphUserCreditCard.template)
+    let mockService = MockService(fetchGraphCreditCardsResponse: response)
+
+    withEnvironment(apiService: mockService, currentUser: User.template) {
+      self.reloadPaymentMethods.assertDidNotEmitValue()
+      self.applePayButtonHidden.assertDidNotEmitValue()
+
+      self.vm.inputs.configureWith((User.template, Project.template, true))
+      self.vm.inputs.viewDidLoad()
+
+      self.scheduler.run()
+
+      self.applePayButtonHidden.assertValues([false])
+      self.reloadPaymentMethods.assertValue(response.me.storedCards.nodes)
+    }
+  }
+
+  func testReloadPaymentMethods_Error_LoggedIn_ApplePayCapable_isFalse() {
```

Comment:
Is there a scenario where we could be `true` for Apple Pay capable but error on fetching the cards? And do we want to test that?

Reply:
Sure